x/mongo/driver/operation_exhaust.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package driver
+
+import (
+	"context"
+	"errors"
+
+	"go.mongodb.org/mongo-driver/x/mongo/driver/description"
+)
+
+// ExecuteExhaust gets a connection from the provided deployment and reads a response from it. This will error if the
+// connection CurrentlyStreaming function returns false.
+func (op Operation) ExecuteExhaust(ctx context.Context, conn Streamer, scratch []byte) error {
```

Comment:
I struggled a bit with the parameters for this function. Even though the callers of this method will use something like `SingleConnectionDeployment, I don't think it makes sense to go through server selection and getting a connection like the normal `Execute` function does because a response can semantically only be streamed from one connection, so I opted to have this take a `Streamer`.

We could add extra validation like confirming that some operation fields like `Deployment`, `Database`, and `CommandFn` aren't set because those are used to get connections or create wire messages, so they aren't relevant here.